src/main/java/com/yelp/nrtsearch/server/luceneserver/DirectoryFactory.java

```diff
@@ -87,19 +92,32 @@ public Directory open(Path path) throws IOException {
         throw new IllegalArgumentException(
             "could not locate Directory sub-class \"" + dirImpl + "\"; verify CLASSPATH");
       }
-      final Constructor<? extends Directory> ctor;
+      Constructor<? extends Directory> ctor = null;
       try {
         ctor = dirClass.getConstructor(Path.class);
-      } catch (NoSuchMethodException nsme) {
+      } catch (NoSuchMethodException ignored) {
+      }
+      try {
+        ctor = dirClass.getConstructor(Path.class, boolean.class);
+      } catch (NoSuchMethodException ignored) {
+      }
+      if (ctor == null) {
```

Comment:
I'm wondering if we should do away with this whole `else` block. I think it would be used only if someone decides to create a custom directory plugin. (Not a blocker for this PR)

Reply:
Yeah, if we really want to support this, we should probably add a plugin interface at some point.